php/src/Google/Protobuf/Internal/Message.php

```diff
@@ -1574,11 +1597,11 @@ private function existField($field)
 
         $values = $this->$getter();
         if ($field->isMap()) {
-            return count($values) !== 0;
+            return $emitDefaults || count($values) !== 0;
         } elseif ($field->isRepeated()) {
-            return count($values) !== 0;
+            return $emitDefaults || count($values) !== 0;
         } else {
-            return $values !== $this->defaultValue($field);
+            return $values !== $this->defaultValue($field) || ($emitDefaults && $this->defaultValue($field) !== null);
```

Comment:
I think this should just be `$emitDefaults || $values !== $this->defaultValue($field)`.

Even if a default is not specified in the `.proto` file, the "default default" is 0 or an empty string. I don't think `$this->defaultValue($field)` should ever return `null`, except possibly message/repeated/map fields, but those should all have been handled above.